syntax = "proto3";

package be.cloud;
import "google/protobuf/empty.proto";

service Scoreboard {
  rpc SendScore(UserScore) returns (google.protobuf.Empty) {}
  rpc GetLeaderboard(LeaderboardRequest) returns (LeaderboardResponse) {}
}

message UserScore {
  string user_id = 1;
  int32 score = 2;
}

message LeaderboardRequest {
}

message LeaderboardResponse {
  repeated UserScore scores = 1;
}
